liferay-lima/liferay-portal
# LPS-112149 Message Boards Advanced Reply Button Styles

This is a possible fix for [Message Boards "Advanced Reply" button link lost its styling](https://issues.liferay.com/browse/LPS-112149)

**Test plan:**
- Navigate to Admin > Message Boards
- Add a new thread
- Add a reply
- In the reply form, look at the "Advanced Reply" button link

The "Advanced Reply" button is **expected to look like a link**

**The fix:**
The new `clay:*` tags contain less implicit logic making them easier to control for the desired final results. This PR simply rewrites the old `aui:button` usage with a new `clay:button` one. 

All logic remains the same since the button action is attached via event listener inside `MBPortlet.es.js`. A future improvement that could be made would be to pass a `propsTransformer` to the `clay:button` to directly assign an `onClick` prop thus avoiding the additional module (at least for that part). Since that's a slightly bigger refactor, leaving it out for now :)

**Before**
<img width="1040" alt="Screen Shot 2020-08-05 at 07 16 21" src="https://user-images.githubusercontent.com/905006/89374413-95f8f180-d6eb-11ea-83e7-c8ff9453c86f.png">

**After**
<img width="1025" alt="Screen Shot 2020-08-05 at 07 15 20" src="https://user-images.githubusercontent.com/905006/89374385-87123f00-d6eb-11ea-83dd-3cfc6308731f.png">

**PS:** There's a `NullPointerExcepction` happening when redirecting to the Advanced Reply view. This seems to be already present in `master`, so you might want to take a look

```
2020-08-05 05:21:03.066 ERROR [http-nio-8080-exec-5][IncludeTag:128] Current URL /group/guest/~/control_panel/manage?p_p_id=com_liferay_message_boards_web_portlet_MBAdminPortlet&p_p_lifecycle=0&p_p_state=maximized&p_p_mode=view&_com_liferay_message_boards_web_portlet_MBAdminPortlet_mvcRenderCommandName=%2Fmessage_boards%2Fedit_message&_com_liferay_message_boards_web_portlet_MBAdminPortlet_redirect=http%3A%2F%2Flocalhost%3A8080%2Fgroup%2Fguest%2F~%2Fcontrol_panel%2Fmanage%3Fp_p_id%3Dcom_liferay_message_boards_web_portlet_MBAdminPortlet%26p_p_lifecycle%3D0%26p_p_state%3Dmaximized%26p_p_mode%3Dview%26_com_liferay_message_boards_web_portlet_MBAdminPortlet_mvcRenderCommandName%3D%252Fmessage_boards%252Fview_message%26_com_liferay_message_boards_web_portlet_MBAdminPortlet_messageId%3D38207&_com_liferay_message_boards_web_portlet_MBAdminPortlet_threadId=38208&_com_liferay_message_boards_web_portlet_MBAdminPortlet_mbCategoryId=0&_com_liferay_message_boards_web_portlet_MBAdminPortlet_parentMessageId=38207&_com_liferay_message_boards_web_portlet_MBAdminPortlet_priority=0.0 generates exception: java.lang.NullPointerException
java.lang.NullPointerException
    at org.apache.jsp.message_005fboards.view_005fthread_005fmessage_jsp._jspService(view_005fthread_005fmessage_jsp.java:486)
    [...]
```
